File: C++/pe016.cpp
#include <iostream>
#include <gmpxx.h>


int main() {
   
    mpz_class Num = 2;

for (int i = 1; i < 15; i++)
{
    Num *= 2;
}



mpz_class Sum = 0;

while (Num >= 1)
{
    Sum += Num%10;
    Num /=10; 

}

std::cout << Sum << std::endl;

     

    return 0;
}

File: Python/pe016.py

# Problem 16
# ==========
#    2^15 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.
#    What is the sum of the digits of the number 2^1000?

num = str(2**1000)
print(sum(int(n) for n in str(num)))